inline_Ob.cc

```diff
@@ -359,13 +359,37 @@ namespace Chroma
         OC0 *= 0.0;
         OD *= 0.0;
         OD0 *= 0.0;
-
-
-
         
+        //manually do the z=2 operators
+        u2 = shift(u[z], FORWARD, z);
+        un = u[z]*u2;
+        for(int mu = 0; mu < Nd; mu++)
+        {
+            for(int nu = mu+1; nu < Nd; nu++)
+            {
+                Fn[nu][mu] = shift(shift(F[nu][mu], FORWARD, z), FORWARD, z);
+                Fn[mu][nu] = -Fn[nu][mu];
+            }
+        }
+        //manually do the z=1 operators, and check the shift function
+/*
+        u2 = shift(shift(u[z], FORWARD, z), BACKWARD, z);
+        un = u2;
+        for(int mu = 0; mu < Nd; mu++)
+        {
+            for(int nu = mu+1; nu < Nd; nu++)
+            {
+                Fn[nu][mu] = shift(shift(shift(F[nu][mu], FORWARD, z), FORWARD, z), BACKWARD,z);
+                Fn[mu][nu] = -Fn[nu][mu];
+            }
+        }
+*/
+
+		
         for(int i=0;i<4; i++)
         {
             O3 += F[z][i]*un*Fn[z][i]*adj(un);
+            //O3 += F[z][i]*un*Fn[z][i]*adj(u2)*adj(u[z]);
```

Comment:
Check the adj function, results in directory "/mnt/home/fanzhouy/fanzhouynew/gluon_debug"